modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseUpgradeMatcherReplacementCheck.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nícolas Moura
+ */
+public abstract class BaseUpgradeMatcherReplacementCheck
+	extends BaseUpgradeCheck {
+
+	protected String beforeFormatIteration(
+		String fileName, String absolutePath, String content) {
+
+		return content;
+	}
+
+	@Override
+	protected String format(
+		String fileName, String absolutePath, String content) {
+
+		String newContent = beforeFormatIteration(
+			fileName, absolutePath, content);
+
+		Matcher matcher = getPattern().matcher(content);
```

Comment:
Method chaining is not allowed (though it seems, for some reason, normal SF is not catching this). Please split this up.

Reply:
Hi Kevin, I believe that SF allows one level of chaining Because of that I did this way it's an error on the chaining check so?

Reply:
I think it is an error. Usually, Brian Chan does not like seeing this type of method chaining unless we are dealing with a builder class or something similar. It might be okay with him.

Just to be safe, I would break this up.